src/main/pg/rx.c

```diff
@@ -48,11 +48,19 @@
 
 #endif
 
+#ifndef SERIALRX_HALFDUPLEX
+#if defined(USE_SERIALRX_FPORT) || defined(USE_SERIALRX_SRXL2) || defined(USE_SERIALRX_GHST)
+#define SERIALRX_HALFDUPLEX 1
+#else
+#define SERIALRX_HALFDUPLEX 0
+#endif
+#endif
+
 PG_REGISTER_WITH_RESET_FN(rxConfig_t, rxConfig, PG_RX_CONFIG, 4);
 void pgResetFn_rxConfig(rxConfig_t *rxConfig)
 {
     RESET_CONFIG_2(rxConfig_t, rxConfig,
-        .halfDuplex = 0,
+        .halfDuplex = SERIALRX_HALFDUPLEX,
```

Comment:
I think this is the wrong place to do this. I think it would be better to add a "AUTO, ON, OFF" and under AUTO and protocol initialisation it is decided whether it is on or off. The reason is it seems to follow the protocol selection.

Cloud build this would be "ok", but if more than one protocol is added, then it becomes problematic.

Reply:
Updated the condition and moved to common_post.h.

Reply:
No, it would work with cloud build but not with core build. We have to move the logic to protocol initialization.